youtube_dl/extractor/youmaker.py

```diff
@@ -0,0 +1,144 @@
+            },
+        },
+    ]
+
+    def __init__(self, downloader=None):
+        """Constructor. Receives an optional downloader."""
+        super(YouMakerIE, self).__init__(downloader=downloader)
+        self.__protocol = "https"
+
+    def _fix_url(self, url):
+        if url.startswith("//"):
+            return "%s:%s" % (self.__protocol, url)
+        return url
+
+    @property
+    def _base_url(self):
+        return self._fix_url("//www.youmaker.com")
+
+    @property
+    def _api_url(self):
+        return "%s/v1/api" % self._base_url
+
+    @property
+    def _asset_url(self):
+        # as this url might change in the future
+        # it needs to be extracted from some js magic...
+        return self._fix_url("//vs.youmaker.com/assets")
+
+    def _extract_subtitles(self, system_id=None):
+        subtitles = {}
+
+        if system_id is None:
+            return subtitles
+
+        info = self._download_json(
+            "%s/video/subtitle?systemid=%s" % (self._api_url, system_id),
+            system_id,
+        )
+
+        if info.get("status") != "ok":
+            return subtitles
+
+        for item in info.get("data", []):
+            subtitles.setdefault(item["language_code"], []).append(
+                {"url": "%s/%s" % (self._asset_url, item["url"])}
+            )
+        return subtitles
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        if url.startswith("http://"):
+            self.__protocol = "http"
+        info = self._download_json(
+            "%s/video/metadata/%s" % (self._api_url, video_id), video_id
+        )
+
+        status = info.get("status", "something went wrong")
+        if status != "ok":
+            raise utils.ExtractorError(status, expected=True)
+
+        info = info["data"]
+        info["tags"] = [
+            tag.strip() for tag in info.get("tag", "").strip("[]").split(",") if tag
+        ]
+        info["channel_url"] = (
+            "%s/channel/%s" % (self._base_url, info["channel_uid"])
+            if "channel_uid" in info
+            else None
+        )
+        video_info = info.get("data", {})
+        duration = video_info.get("duration")
+        formats = []
+        playlist = video_info.get("videoAssets", {}).get("Stream")
+
+        if playlist:
+            formats.extend(
+                self._extract_m3u8_formats(
+                    self._fix_url(playlist),
+                    video_id,
+                    ext="mp4",
+                )
+            )
+
+        if formats:
+            self._sort_formats(formats)
+            for item in formats:
+                item["format_id"] = "hls-%dp" % item["height"]
+                if duration and item.get("tbr"):
+                    item["filesize_approx"] = 128 * item["tbr"] * duration
+
+        return {
+            "id": info["video_uid"],
+            "title": info["title"],
+            "description": info["description"],
```

Comment:
`info.get('description')` ? Not a mandatory item. Similarly with `info['uploaded_at']` below.

Reply:
Ok. Please see new commits...